### ISSUE TITLE
Parse error on GRANT T-SQL

### ISSUE THREAD

**Issue by grv87** [sqlfluff/sqlfluff#7893]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Problematic SQL:
```sql
GRANT VIEW DEFINITION
    ON OBJECT::[some_schema].[some_object] TO [some_user];
```

### Expected Behaviour

Code is parsed

### Observed Behaviour

```
Unable to lint ....sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
RuntimeError: Grammar refers to the 'CHANGE' keyword which was not found in the tsql dialect.

The syntax in the query is not (yet?) supported. Try to narrow down your query to a minimal, reproducible case and raise an issue on GitHub.

Or, even better, see this guide on how to help contribute keyword and/or dialect updates:
https://docs.sqlfluff.com/en/stable/perma/contribute_dialect_keywords.html

Traceback (most recent call last):
File "C:\Python312\Lib\site-packages\sqlfluff\core\linter\runner.py", line 206, in run  lint_result.reraise()File "C:\Python312\Lib\site-packages\sqlfluff\core\linter\runner.py", line 367, in reraise  raise self.ee.with_traceback(self.tb)File "C:\Python312\Lib\site-packages\sqlfluff\core\linter\runner.py", line 251, in _apply  return Linter.lint_rendered(rendered, rule_pack, task.fix, None)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\linter\linter.py", line 905, in lint_rendered  parsed = cls.parse_rendered(rendered)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\linter\linter.py", line 415, in parse_rendered  parsed, parse_errors = cls._parse_tokens(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\linter\linter.py", line 286, in _parse_tokens  parsed: Optional[BaseSegment] = parser.parse(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\parser.py", line 51, in parse  root = self.RootSegment.root_parse(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\segments\file.py", line 87, in root_parse  match = cls.match_grammar.match(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\sequence.py", line 237, in match  elem_match = elem.match(segments[:max_idx], _idx, ctx)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\anyof.py", line 196, in match  match, matched_option = longest_match(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\match_algorithms.py", line 235, in longest_match  res_match = matcher.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\base.py", line 427, in match  return elem.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\segments\base.py", line 697, in match  match = cls.match_grammar.match(segments, idx, ctx)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\anyof.py", line 196, in match  match, matched_option = longest_match(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\match_algorithms.py", line 235, in longest_match  res_match = matcher.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\sequence.py", line 237, in match  elem_match = elem.match(segments[:max_idx], _idx, ctx)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\base.py", line 427, in match  return elem.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\anyof.py", line 196, in match  match, matched_option = longest_match(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\match_algorithms.py", line 235, in longest_match  res_match = matcher.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\base.py", line 427, in match  return elem.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\anyof.py", line 196, in match  match, matched_option = longest_match(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\match_algorithms.py", line 235, in longest_match  res_match = matcher.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\sequence.py", line 237, in match  elem_match = elem.match(segments[:max_idx], _idx, ctx)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\base.py", line 427, in match  return elem.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\segments\base.py", line 697, in match  match = cls.match_grammar.match(segments, idx, ctx)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\anyof.py", line 196, in match  match, matched_option = longest_match(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\match_algorithms.py", line 235, in longest_match  res_match = matcher.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\base.py", line 427, in match  return elem.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\segments\base.py", line 697, in match  match = cls.match_grammar.match(segments, idx, ctx)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\sequence.py", line 237, in match  elem_match = elem.match(segments[:max_idx], _idx, ctx)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\anyof.py", line 196, in match  match, matched_option = longest_match(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\match_algorithms.py", line 235, in longest_match  res_match = matcher.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\base.py", line 427, in match  return elem.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\segments\base.py", line 697, in match  match = cls.match_grammar.match(segments, idx, ctx)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\delimited.py", line 148, in match  match, _ = longest_match(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\match_algorithms.py", line 235, in longest_match  res_match = matcher.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\sequence.py", line 237, in match  elem_match = elem.match(segments[:max_idx], _idx, ctx)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\anyof.py", line 196, in match  match, matched_option = longest_match(  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\match_algorithms.py", line 235, in longest_match  res_match = matcher.match(segments, idx, parse_context)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\sequence.py", line 237, in match  elem_match = elem.match(segments[:max_idx], _idx, ctx)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\base.py", line 407, in match  elem = self._get_elem(dialect=parse_context.dialect)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\parser\grammar\base.py", line 376, in _get_elem  return dialect.ref(self._ref)  ^^^^^^^^^^^^^^^^^File "C:\Python312\Lib\site-packages\sqlfluff\core\dialects\base.py", line 320, in ref  raise RuntimeError(  ^^^^^^^^^^^^^^^^^
```

### How to reproduce

`sqlfluff lint test.sql`

### Dialect

tsql

### Version

4.2.1

### Configuration

[sqlfluff]
dialect = tsql

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
